packages/http/src/validator/__tests__/index.spec.ts

```diff
@@ -51,150 +43,112 @@ describe('HttpValidator', () => {
                 responses: [{ code: '200' }],
               },
-              extendResource,
+              3,
             ),
-            input: { method: 'get', url: { path: '/' } },
-            config: { cors: false, mock: { dynamic: false }, validate: { request: { headers: true } } },
-          }),
-        ).toEqual([mockError]);
-
-        expect(resolveValidationConfigModule.resolveRequestValidationConfig).toHaveBeenCalled();
-        expect(httpBodyValidator.validate).not.toHaveBeenCalled();
-        expect(httpHeadersValidator.validate).toHaveBeenCalledWith({}, [], undefined);
-        expect(httpQueryValidator.validate).not.toHaveBeenCalled();
-      };
-
-      describe('request is not set', () => {
-        it('validates headers', test());
-      });
-
-      describe('request is set', () => {
-        describe('request.headers is not set', () => {
-          it('validates headers', test({ request: { path: [], query: [], cookie: [], headers: [] } }));
-        });
-
-        describe('request.headers is set', () => {
-          it('validates headers', test({ request: { path: [], query: [], cookie: [], headers: [] } }));
+          );
         });
       });
     });
+  });
 
-    describe('query validation in enabled', () => {
-      const test = (extendResource?: Partial<IHttpOperation>, extendInput?: Partial<IHttpRequest>) => () => {
-        jest
-          .spyOn(resolveValidationConfigModule, 'resolveRequestValidationConfig')
-          .mockReturnValueOnce({ body: false, headers: false, hijack: false, query: true });
-
-        expect(
-          httpValidator.validateInput({
-            resource: Object.assign(
-              {
-                method: 'get',
-                path: '/',
-                id: '1',
-                request: {},
-                responses: [{ code: '200' }],
-              },
-              extendResource,
-            ),
-            input: Object.assign({ method: 'get', url: { path: '/', query: {} } }, extendInput),
-            config: { cors: false, mock: { dynamic: false }, validate: { request: { query: true } } },
-          }),
-        ).toEqual([mockError]);
+  describe('headers validation in enabled', () => {
+    const test = (extendResource?: Partial<IHttpOperation>, length: number = 1) => () => {
+      expect(
+        httpValidator.validateInput({
+          resource: Object.assign(
+            {
+              method: 'get',
+              path: '/',
+              id: '1',
+              request: {},
+              responses: [{ code: '200' }],
+            },
+            extendResource,
+          ),
+          input: { method: 'get', url: { path: '/' } },
+          config: { cors: false, mock: { dynamic: false }, validateRequest: true, validateResponse: true },
+        }),
+      ).toHaveLength(length);
+    };
+
+    describe('request is not set', () => {
+      it('validates headers', test(undefined, 2));
```

Comment:
this is a bit enigmatic, can we add some info here? Like, what's the reason for an undefined request to have 2 not-passed checks?